### DISCUSSION TITLE
How to use CSR: client-side rendering

### DISCUSSION THREAD

**Discussion by japboy** [i18next/next-i18next#2142] [Q&A] [answered]
This library seems to need to use `serverSideTranslations` to decide initial locale, which means I have to use SSG/SSR. Is my understanding correct?

Is there a way to decide the initial locale on CSR: client-side rendering? Like, on the server-side response, it returns empty page (like skeleton UI), and on the client-side, web browser decides the locale somehow, loads the language file and (re-)renders the page.

I need to cover SSG/SSR, and CSR situation. Thanks.

**Comment by adrai** [comment, marked as answer]
https://github.com/i18next/next-i18next#client-side-loading-of-translations-via-http
https://github.com/i18next/i18next-http-backend/tree/master/example/next
https://locize.com/blog/next-i18next/
client side only example: https://github.com/i18next/i18next-http-backend/blob/master/example/next/pages/client-page.js
client side "refreshed" example: https://github.com/i18next/i18next-http-backend/blob/master/example/next/pages/lazy-reload-page.js

**Reply by japboy** [reply]
Thank you! This should be something I was looking for ✨
